File: java/ValidAnagram.java
import java.util.Arrays;

class Solution {
  /**
   * Function that checks whether any two strings are anagrams.
   * @param s the first string.
   * @param t the second string.
   * @return whether the strings are the same.
   */
  public boolean isAnagram(String s, String t) {
    // Trim both strings to remove unecessary information. 
    s.trim();
    t.trim();

    // Convert both into char arrays to manipulate them easier.
    char[] sArray = s.toCharArray();
    char[] tArray = t.toCharArray();

    // Sort the arrays so the characters are the same at all spots in the array.
    Arrays.sort(sArray);
    Arrays.sort(tArray);

    // If the arrays are not equal in their content, return false, otherwise return true. 
    return Arrays.equals(sArray, tArray);
  }
}

File: python/ValidAnagram.py
class Solution:

    # Function that checks whether any two strings are anagrams.
    def is_anagram(self, s: str, t: str) -> bool:

        # Base case, s and t are of different lengths they cannot have the same letters therefore impossible to rearrange.
        if len(s) != len(t):
            return False

        # Strip of extra whitespace
        s.strip()
        t.strip()

        # Convert into arrays and sort, this will allow to check for array equality rapidly. 
        sArray = list(s)
        tArray = list(t)
        sArray.sort()
        tArray.sort()

        if sArray == tArray:
            return True
